keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -86,17 +69,28 @@ class ArtifactListener(
   @EventListener(ArtifactRegisteredEvent::class)
   fun onArtifactRegisteredEvent(event: ArtifactRegisteredEvent) {
     val artifact = event.artifact
+    val deliveryConfig = artifact.deliveryConfigName
+      ?.let { repository.getDeliveryConfig(it) }
+      ?: error("Delivery config name missing in artifact")
```

Comment:
Should we be throwing spinnaker exceptions instead of using `IllegalStateException` here?

Reply:
(comment applies for every use of `error()` in this PR

Reply:
Good point. Let me create a sub-class of `SystemException` for these cases, because I couldn't find a good one in kork.